object.go

```diff
@@ -0,0 +1,12 @@
+package logical
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Annotation defines the logical annotation that can be attached to a Kubernetes object
+// to determine the logical cluster name the object belongs to.
+const Annotation = "logical.sigs.x-k8s.io/cluster"
+
+// NameFromObject returns the logical cluster name from the given Kubernetes object, if any.
+func NameFromObject(obj metav1.Object) Name {
```

Comment:
Would not use metav1.Object to minimize dependencies. Suggest instead:

```go
type AnnotatedObject interface {
  GetAnnotations() map[string]string
}
```

Feel free to use some other name besides AnnotatedObject 😄

Reply:
Nice and lean now